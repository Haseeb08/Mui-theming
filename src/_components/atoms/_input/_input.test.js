import React from "react";
import { create } from "react-test-renderer";
import { render } from "@testing-library/react";
//import "@testing-library/jest-dom/extend-expect";
import Input from "./_input";

describe("Input testing", () => {
  test("Matches the snapshot", () => {
    const input = create(<Input />);
    expect(input.toJSON()).toMatchSnapshot();
  });
  test("should have input bar", () => {
    const { getByTestId } = render(<Input />);
    expect(getByTestId("input")).toBeInTheDocument();
  });
  test("should be have default value, if prop is not provided", () => {
    const { getByTestId } = render(<Input />);
    expect(getByTestId("input")).toHaveTextContent("1+2+3");
  });
  test("should be have input value from props", () => {
    const { getByTestId } = render(<Input input="1234"/>);
    expect(getByTestId("input")).toHaveTextContent("1234");
  });
  
});