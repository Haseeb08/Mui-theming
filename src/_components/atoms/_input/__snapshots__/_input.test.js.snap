// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input testing Matches the snapshot 1`] = `
<div
  className="makeStyles-input-1"
  data-testid="input"
>
  1+2+3
</div>
`;
